src/providers/DocumentContentProvider.ts

```diff
@@ -132,8 +131,9 @@ export class DocumentContentProvider implements vscode.TextDocumentContentProvid
         });
       }
     }
+    const params = new URLSearchParams(uri.query);
     if (namespace && namespace !== "") {
-      if (isCsp) {
+      if (isCsp && !params.has("csp")) {
```

Comment:
Is the extra condition necessary? Won't it also mean that if wFolderUri already specified `?csp=1` (say) and a value for `namespace` was passed, the uri returned by this function will lack the `csp=1` queryparam?

Reply:
Yes, you're correct. I'll fix it